File: php/src/LoggerService/NullLogger.php
<?php

declare(strict_types=1);

namespace FactChecker\LoggerService;

class NullLogger implements Logger
{
    public function emergency($message, array $context = [])
    {
        return null;
    }

    public function alert($message, array $context = [])
    {
        return null;
    }

    public function critical($message, array $context = [])
    {
        return null;
    }

    public function error($message, array $context = [])
    {
        return null;
    }

    public function warning($message, array $context = [])
    {
        return null;
    }

    public function notice($message, array $context = [])
    {
        return null;
    }

    public function info($message, array $context = [])
    {
        return null;
    }

    public function debug($message, array $context = [])
    {
        return null;
    }

    public function log($level, $message, array $context = [])
    {
        return null;
    }
}

File: java/app/src/main/java/factchecker/LoggerService/NullLogger.java
package factchecker.LoggerService;

public class NullLogger implements Logger {
    @Override
    public void fatal(String message) {
    }

    @Override
    public void error(String message) {
    }

    @Override
    public void warn(String message) {
    }

    @Override
    public void info(String message) {
    }

    @Override
    public void debug(String message) {
    }

    @Override
    public void trace(String message) {
    }

    @Override
    public void setLevel(Level level) {
    }

    @Override
    public void log(Level level, String message) {
    }
}
